test/passing/comments.ml

```diff
@@ -129,3 +129,22 @@ type t = {
 a : int [@default a]
 (* comment *)
 ;  b : flag }
+
+;;
+match (* comment about entire pattern matching *)
+      e with k -> z
```

Comment:
Note that in
```
(* comment about entire pattern matching *)
match e with k -> z
```
the comment comes before `match` and that should be preserved. It is not enough to always put the comments before or after the `match`, the original position needs to be preserved.

Reply:
Agreed, but for the record, this is not a regression, this is also formatted like this on master.

Reply:
This gets formatted as:
```
(* comment about entire pattern matching *)

;;
match e with k -> z
```
as it is the case for any expression on master. Is this what you would like to change? It would belong in another PR.

Reply:
The whole aim of #341 is to make sure that a comment before a token, here `match`, are still before the token after being formatted, and vice versa: a comment after a token stays after it. Details about `;;` are probably tangential, and it might be better to use test cases which are bodies of let-bindings.